staging/src/k8s.io/apiserver/pkg/server/filters/goaway_test.go

```diff
@@ -202,7 +202,7 @@ func newGOAWAYClient(dial func(network, addr string, cfg *tls.Config) (net.Conn,
 		Proxy:               http.ProxyFromEnvironment,
 		TLSHandshakeTimeout: 10 * time.Second,
 		TLSClientConfig:     tlsConfig,
-		MaxIdleConnsPerHost: 25,
+		MaxIdleConnsPerHost: -1,
```

Comment:
We can enable client pool for the case TestGOAWAYConcurrency.

Reply:
this function `newGOAWAYClient` is only used for `TestGOAWAYConcurrency` and `TestClientReceivedGOAWAY`, so I think is better to create the http client in each test than adding more parameters to this function, I think there is no benefit in sharing it between the tests.

Reply:
ok.